feature_detection/scripts/Hough_Transform_orientation_based.py

```diff
@@ -0,0 +1,352 @@
+import cv2
+import numpy as np
+from copy import deepcopy
+
+
+""" 
+ToDo
++ tuning of the parameter of cv2.HoughLinesP makes it more robust, but less sensitive
++ component detection --> some kind of filter(?) has to be applied
++ idea on how to improve the algorithm: use eucledian norm to get distance of two lines to get independent of orientation
+"""
+
+# feature detection based on hough transform
+class HoughTransform:
+    """
+    Class used in feature detection to detect shapes of objects consisting of straight lines. Returns the bounding boxes and the centroids of them.
+
+    """
+
+    @staticmethod
+    def lines_coord(lines_HT, orientation, e=8):
+        """
```

Comment:
Make orientation a default arg?

```suggestion
    @staticmethod
    def lines_coord(lines_HT, orientation=0, e=8):
        """
```